tools/acl_compressor/sources/main.cpp

```diff
@@ -100,30 +102,44 @@ struct RawOutputWriterImpl : public acl::OutputWriter
 
 struct Options
 {
+	const char*		input_filename;
 	bool			output_stats;
 	const char*		output_stats_filename;
 
 	Options()
-		: output_stats(false)
+		: input_filename(nullptr),
+		  output_stats(false)
 		, output_stats_filename(nullptr)
 	{}
 };
 
-static Options parse_options(int argc, char** argv)
+static bool parse_options(int argc, char** argv, Options& options)
 {
-	Options options;
-
 	for (int arg_index = 0; arg_index < argc; ++arg_index)
 	{
 		char* argument = argv[arg_index];
+		
+		if (std::strcmp(argument, "-file") == 0)
+		{
+			++arg_index;
+			if (arg_index >= argc)
+			{
+				printf("-file must be followed by a filename\n");
```

Comment:
I prefer arguments of the form -file=./foo/bar.txt that way you don't need to validate things like this, see -stats below